File: php/tests/Unit/Domain/Rules/ValueObject/TargetTypeTest.php
<?php

declare(strict_types=1);

namespace Tests\Unit\Domain\Rules\ValueObject;

use Alxarafe\App\Domain\Rules\ValueObject\TargetType;
use PHPUnit\Framework\TestCase;

final class TargetTypeTest extends TestCase
{
    public function testValues(): void
    {
        self::assertSame('LOCATION', TargetType::LOCATION->value);
        self::assertSame('FAMILY', TargetType::FAMILY->value);
    }
}

File: java/src/test/java/com/alxarafe/domain/rules/valueobject/TargetTypeTest.java
package com.alxarafe.domain.rules.valueobject;

import com.alxarafe.app.domain.rules.valueobject.TargetType;
import org.junit.jupiter.api.Test;

import static org.junit.jupiter.api.Assertions.*;

class TargetTypeTest {

    @Test
    void values() {
        assertEquals(TargetType.LOCATION, TargetType.valueOf("LOCATION"));
        assertEquals(TargetType.FAMILY, TargetType.valueOf("FAMILY"));
        assertEquals(2, TargetType.values().length);
    }
}
